x_transformers/x_transformers.py

```diff
@@ -2106,14 +2122,26 @@ def forward(
 
             x = x[:, :n]
 
-        # projecting to logits
+        # cls token or pooling for classification
+        if self.use_cls:
+            x = x[:, 0]
+        
+        if self.use_pooling:
+            x = self.pooling(x).squeeze()
```

Comment:
for the pooling, we need to account for masking (masked averaging)

Reply:
i can take care of this if you'd like, it is all around a bit tricky

Reply:
Yes, please do so. Thank you 👍